precode_test.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+//Нужно реализовать три теста:
+
+// 1) Запрос сформирован корректно, сервис возвращает код ответа 200
+// и тело ответа не пустое.
+func TestMainHandlerWhenOk(t *testing.T) {
+	req := httptest.NewRequest("GET", "/cafe?count=2&city=moscow", nil)
+
+	responseRecorder := httptest.NewRecorder()
+	handler := http.HandlerFunc(mainHandle)
+	handler.ServeHTTP(responseRecorder, req)
+
+	if status := responseRecorder.Code; status != http.StatusOK {
+		t.Errorf("expected status code: %d, got %d", http.StatusOK, status)
+	}
+}
+
+// 2) Город, который передаётся в параметре city, не поддерживается.
+// Сервис возвращает код ответа 400 и ошибку wrong city value в теле ответа.
+func TestMainHandlerWhenUnsupportedCity(t *testing.T) {
+	req, err := http.NewRequest("GET", "/cafe?count=3&city=tula", nil)
+	require.NoError(t, err, "Ошибка при создании запроса!")
+
+	responseRecorder := httptest.NewRecorder()
+
+	handler := http.HandlerFunc(mainHandle)
+	handler.ServeHTTP(responseRecorder, req)
+
+	require.Equal(t, http.StatusBadRequest, responseRecorder.Code, "Код ответа 400")
```

Comment:
Последним аргументом пишется ошибка, что не так
```suggestion
	require.Equal(t, http.StatusBadRequest, responseRecorder.Code, "Код ответа не 400")
```